auth/src/express/middleware/route-service.ts

```diff
@@ -78,7 +78,17 @@ export default class RouteService extends RouteHandler {
 
     public hash(request: express.Request, response: express.Response): void {
         const toHash = request.body['toHash'];
-        this.sendResponse(true, this.authHandler.toHash(toHash), response);
+        const salt = request.body['salt'];
+        this.sendResponse(true, 'Successful', response, 200, { hash: this.authHandler.toHash(toHash, salt) });
+    }
+
+    public isEquals(request: express.Request, response: express.Response): void {
+        const toHash = request.body['toHash'];
+        const toCompare = request.body['toCompare'];
+        const salt = request.body['salt'];
```

Comment:
Do not expect the customer to provide the salt. Get it from toCompare. E.g. the backend does not know about the structure of the hashes, only you know it!